### ISSUE TITLE
Challenge Solution in "Updating Arrays in State" Uses Mutation

### ISSUE THREAD

**Issue by EdZ543** [reactjs/reactjs.org#5672]
For the solution to the third challenge in the [Updating Arrays in State](https://beta.reactjs.org/learn/updating-arrays-in-state#updating-objects-inside-arrays) page, it uses the following function:

```javascript
function handleAddTodo(title) {
  setTodos([
    ...todos,
    {
      id: nextId++,
      title: title,
      done: false
    }
  ]);
}
```

This changes `nextId`, a variable outside the component's scope, which I believe is a mutation. It should instead use the length of `todos` to set the next id.

**Comment by jaisaichand** [comment]
Pls assign this to me, I would like to work on this if no one is currently working on it

**Comment by Ashish-simpleCoder** [comment]
We can also try using the Date.now().toString() as value for the **id** attribute. 
It will always be unique and safe.

**Comment by rickhanlonii** [comment]
Yeah, the mutation is ok because it's mutating a module level variable in a click handler. But since this is teaching you to not mutated, it would be good to include a note in the solution that explains why mutating the nextId is OK.

**Comment by rickhanlonii** [comment]
If you're interested in approving it, please just submit a PR, we don't assign issues.
